hyperspy/tests/signal/test_2D_tools.py

```diff
@@ -140,6 +142,84 @@ def test_add_ramp_lazy():
     s.add_ramp(-1, -1, -4)
     npt.assert_almost_equal(s.data.compute(), 0)
 
+
+peak_methods = ['skimage', 'max', 'minmax', 'zaefferer', 'stat', 'laplacian_of_gaussians', 'difference_of_gaussians']
+
+coefficients = np.array(
+    [350949.04890400 + 0.j, -22003.98742841 + 51494.56650429j,
+     37292.52741553 + 38067.97686711j, 37292.52741553 - 38067.97686711j,
+     -22003.98742841 - 51494.56650429j]
+)
+coordinates = np.array([[0, 26, 30, 994, 998],
+                        [0, 1003, 39, 985, 21]]
+)
+dense = np.zeros((1024, 1024), dtype=complex)
+dense[coordinates[0], coordinates[1]] = coefficients
+dense = hs.signals.Signal2D(np.real(np.fft.ifft2(dense)))
+dense = dense.isig[500:550, 500:550]
+
+coefficients = np.array(
+    [10, 5, 86, 221, 6, 95, 70, 12, 255, 5, 255, 3, 23,
+     24, 77, 255, 11, 255, 8, 35, 195, 165, 27, 255, 8, 14,
+     255, 21, 53, 107, 255, 18, 255, 4, 26, 255, 39, 27, 255,
+     6, 255, 7, 13, 37, 35, 9, 83]
+)
+coordinates = np.array(
+    [[3, 40],    [3, 138],  [9, 67],   [14, 95],   [20, 23],
+     [20, 122],  [26, 51],  [26, 100], [31, 78],   [31, 128],
+     [37, 107],  [38, 7],   [43, 34],  [43, 84],   [43, 134],
+     [49, 62],   [49, 112], [54, 90],  [60, 17],   [60, 67],
+     [60, 118],  [66, 45],  [66, 96],  [72, 73],   [72, 124],
+     [77, 51],   [77, 101], [83, 28],  [83, 79],   [83, 130],
+     [89, 57],   [89, 107], [95, 85],  [101, 12],  [101, 62],
+     [101, 113], [106, 40], [107, 91], [112, 68],  [113, 119],
+     [119, 97],  [124, 23], [124, 74], [124, 125], [130, 51],
+     [130, 103], [136, 80]])
+sparse = np.zeros((144, 144))
+xs, ys = np.ogrid[:144, :144]
+for (x0, y0), a in zip(coordinates, coefficients):
+    sparse += a * sp.stats.norm.pdf(xs, x0)*sp.stats.norm.pdf(ys, y0)
+sparse = sparse[50:100, 50:100]
+sparse0d = hs.signals.Signal2D(sparse)
+sparse1d = hs.signals.Signal2D(np.array([sparse for i in range(2)]))
+sparse2d = hs.signals.Signal2D(np.array([[sparse for i in range(2)] for j in range(2)]))
+xref, yref = 72, 72
+ref = np.zeros((144, 144))
+ref += 100 * sp.stats.norm.pdf(xs, xref)*sp.stats.norm.pdf(ys, yref)
+
+ref = hs.signals.Signal2D(ref)
+ans = np.empty((1,), dtype=object)
+ans[0] = np.array([[xref, yref]])
+
+
+datasets = [dense, sparse0d, sparse1d, sparse2d]
+
+@pytest.mark.parametrize('method', peak_methods)
+@pytest.mark.parametrize('dataset', datasets)
+def test_creates_signal(method, dataset):
```

Comment:
Also, to skip the `'stat'` method without `sklearn`, add this in the test:
```python
#@pytest.mark.parametrize(....)
def test_whatever_name(method, ....):
    if method=='stat':
        _ = pytest.importorskip("sklearn")
    # continue with the test
```
Could not find a nice way to do that with `parametrize`, but that should suffice